omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,396 @@
+                }else{
+                    errors.add("identifier souce");
+                }
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                if(parseBoolean(sequential.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(sequential.toString()));
+                }else {
+                    errors.add("sequential");
+                }
+            }
+            if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                if(parseBoolean(refillWithScheduledTask.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(refillWithScheduledTask.toString()));
+                }else {
+                    errors.add("refill");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name.toString());
+            if(identifierSource.getBatchSize() > 0 && identifierSource.getSource() != null){
+                Context.getService(IdentifierSourceService.class).addIdentifiersToPool(identifierSource, Integer.parseInt(batchSize.toString()));
+            }
+
+            savedIdentifierSource =  save(identifierSource);
+        }
+        if(errors.toString().length() > 2) {
+            throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+        }
+        return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);
+    }
+    
+    @Override
+    public Object update(String uuid, SimpleObject updateBody, RequestContext context) throws ResponseException {
+        
+       ArrayList<String> errors = new ArrayList<String>();
+       Object updatedIdentifierSource = null;
+       
```

Comment:
Do we have tests for updating the three types?

Reply:
yes I do: [Here](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-13128b20c5113cca7e0908cfb6ad9a13R85)